Polyhedron/demo/Polyhedron/Polyhedron_demo_vtk_plugin.cpp

```diff
@@ -0,0 +1,169 @@
+#include <QMainWindow>
+#include <QFileDialog>
+#include <QString>
+
+#include <fstream>
+
+#include <boost/graph/graph_traits.hpp>
+#include <boost/unordered_map.hpp>
+
+#include <CGAL/boost/graph/properties.h> 
+#include <CGAL/boost/graph/Euler_operations.h>
+#include <CGAL/boost/graph/graph_traits_Polyhedron_3.h>
+#include <CGAL/property_map.h>
+
+#include <vtkSmartPointer.h>
+#include <vtkPolyDataReader.h>
+#include <vtkXMLPolyDataReader.h>
+#include <vtkPolyData.h>
+#include <vtkIdTypeArray.h>
+#include <vtkFieldData.h>
+#include <vtkFloatArray.h>
+#include <vtkDoubleArray.h>
+#include <vtkCell.h>
+#include <vtkCellData.h>
+
+namespace CGAL{
+
+  template <typename TM, typename VertexMap, typename FaceMap>
+  bool vtkPolyData_to_polygon_mesh(vtkPolyData* poly_data,
+                                   TM& tmesh,
+                                   VertexMap& vertex_map,
+                                   FaceMap& face_map)
+  {
+    typedef typename boost::property_map<TM, CGAL::vertex_point_t>::type VPMap;
+    typedef typename boost::property_map_value<TM, CGAL::vertex_point_t>::type Point_3;
+    typedef typename boost::graph_traits<TM>::vertex_descriptor vertex_descriptor;
+    typedef typename boost::graph_traits<TM>::face_descriptor   face_descriptor;
+
+    VPMap vpmap = get(CGAL::vertex_point, tmesh);
+
+    // get nb of points and cells
+    vtkIdType nb_points = poly_data->GetNumberOfPoints();
+    vtkIdType nb_cells = poly_data->GetNumberOfCells();
+
+    //extract points
+    for (vtkIdType i = 0; i<nb_points; ++i)
+    {
+      double coords[3];
+      poly_data->GetPoint(i, coords);
+
+      vertex_descriptor v = add_vertex(tmesh);
+      put(vpmap, v, Point_3(coords[0], coords[1], coords[2]));
+      vertex_map.insert(std::make_pair(i, v));
+    }
+
+    //extract cells
+    for (vtkIdType i = 0; i<nb_cells; ++i)
+    {
+      vtkCell* cell_ptr = poly_data->GetCell(i);
+
+      vtkIdType nb_vertices = cell_ptr->GetNumberOfPoints();
+      if (nb_vertices != 3){
+        std::cerr << "Error a cell with " << nb_vertices << " found\n";
+        return false;
+      }
+      std::vector<vertex_descriptor> vr;
+      vr.push_back(vertex_map[cell_ptr->GetPointId(0)]);
+      vr.push_back(vertex_map[cell_ptr->GetPointId(1)]);
+      vr.push_back(vertex_map[cell_ptr->GetPointId(2)]);
+
+      face_descriptor f = CGAL::Euler::add_face(vr, tmesh);
+      face_map.insert(std::make_pair(i, f));
+    }
+    return true;
+  }
+}//end namespace CGAL
+
+
+class Polyhedron_demo_vtk_plugin :
+  public QObject,
+  public Polyhedron_demo_io_plugin_interface
+{
+  Q_OBJECT
+  Q_INTERFACES(Polyhedron_demo_io_plugin_interface)
+  Q_PLUGIN_METADATA(IID "com.geometryfactory.PolyhedronDemo.IOPluginInterface/1.0")
+
+public:
+  typedef boost::graph_traits<Polyhedron>::vertex_descriptor vertex_descriptor;
+  typedef boost::graph_traits<Polyhedron>::face_descriptor face_descriptor;
+
+  QString nameFilters() const { return "VTK files (*.vtk, *.vtp)"; }
```

Comment:
`"VTK files"` could be changed to `"VTK PolyData files"`, to emphasis that we do not support the other data structures.